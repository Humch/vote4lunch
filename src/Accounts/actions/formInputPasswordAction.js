export default password => ({
  type: 'FORM_INPUT_PASSWORD',
  password,
});
