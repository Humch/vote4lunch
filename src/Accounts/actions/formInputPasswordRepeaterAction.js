export default passwordRepeater => ({
  type: 'FORM_INPUT_PASSWORD_REPEATER',
  passwordRepeater,
});
