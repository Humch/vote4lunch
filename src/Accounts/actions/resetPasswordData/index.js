import resetPasswordData from './resetPasswordData';

export default resetPasswordData;
