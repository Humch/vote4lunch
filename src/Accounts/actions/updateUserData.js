export default (pseudo, email) => ({
  type: 'UPDATE_USER_DATA',
  pseudo,
  email,
});
