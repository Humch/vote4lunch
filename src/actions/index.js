export const saveVoteData = (id, date, pseudo, email, url) => ({
  type: 'SAVE_VOTE_DATA',
  id,
  date,
  pseudo,
  email,
  url
});

export const resetVoteData = () => ({
  type: 'RESET_VOTE_DATA',
});

export const updateUserData = (pseudo, email) => ({
  type: 'UPDATE_USER_DATA',
  pseudo,
  email,
});
