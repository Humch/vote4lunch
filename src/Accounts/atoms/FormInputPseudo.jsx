import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import { FormGroup, Label, Input } from 'reactstrap';

import { FormInputPseudo } from '../actions';

const Pseudo = ({ pseudo, FormInputPseudo: FormInputPseudoAction }) => (
  <FormGroup>
    <Label className="text-white" for="pseudo">Indique ton pseudo</Label>
    <Input
      type="text"
      name="pseudo"
      id="pseudo"
      value={pseudo}
      onChange={e => FormInputPseudoAction(e.target.value)}
      required
    />
  </FormGroup>
);

Pseudo.propTypes = {
  pseudo: PropTypes.string.isRequired,
  FormInputPseudo: PropTypes.func.isRequired,
};

const mstp = ({ voteDataForm }) => ({
  pseudo: voteDataForm.pseudo,
});

const mdtp = dispatch => bindActionCreators({ FormInputPseudo }, dispatch);

export default connect(mstp, mdtp)(Pseudo);
