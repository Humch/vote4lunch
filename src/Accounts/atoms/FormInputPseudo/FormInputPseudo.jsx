import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import { formInputPseudo } from '../../actions';

const FormInputPseudo = ({ pseudo, exist, formInputPseudo: formInputPseudoAction }) => {
  let renderExist = '';

  if (exist) {
    renderExist = 'is-invalid';
  }
  return (
    <input
      type="text"
      name="pseudo"
      id="pseudo"
      value={pseudo}
      onChange={e => formInputPseudoAction(e.target.value)}
      required
      className={`form-control ${renderExist}`}
    />
  );
};

FormInputPseudo.propTypes = {
  pseudo: PropTypes.string.isRequired,
  exist: PropTypes.bool.isRequired,
  formInputPseudo: PropTypes.func.isRequired,
};

const mstp = ({ userData, pseudoChecker }) => ({
  pseudo: userData.pseudo,
  exist: pseudoChecker.payload.exist
});

const mdtp = dispatch => bindActionCreators({ formInputPseudo }, dispatch);

export default connect(mstp, mdtp)(FormInputPseudo);
