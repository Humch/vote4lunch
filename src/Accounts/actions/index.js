import { addUser } from './addUser/addUser';
import formInputPseudo from './formInputPseudoAction';
import formInputPassword from './formInputPasswordAction';
import formInputPasswordRepeater from './formInputPasswordRepeaterAction';
import formInputEmail from './formInputEmailAction';
import { checkPseudo } from './pseudoChecker/pseudoChecker';
import { checkEmail } from './emailChecker/emailChecker';
import updateUserData from './updateUserData';

export {
  addUser,
  formInputPseudo,
  formInputPassword,
  formInputPasswordRepeater,
  formInputEmail,
  checkPseudo,
  checkEmail,
  updateUserData,
};
