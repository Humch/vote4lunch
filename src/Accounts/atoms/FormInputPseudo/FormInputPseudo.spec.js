/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import FormInputPseudo from './FormInputPseudo';

import { formInputPseudo } from '../../actions';

describe('FormInputPseudo Snapshot', () => {
  const initialState = {
    userData: {
      pseudo: 'bob',
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormInputPseudo', () => {
    const renderedValue = renderer.create(<FormInputPseudo store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});


describe('FormInputPseudo (Shallow + passing the {store} directly)', () => {
  const initialState = {
    userData: {
      pseudo: 'bob',
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<FormInputPseudo store={store} />);
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('pseudo')).toEqual(initialState.userData.pseudo);
  });
});

describe('FormInputPseudo - REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = {
    userData: {
      pseudo: 'bob',
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    store.clearActions();
    wrapper = mount(<Provider store={store}><FormInputPseudo /></Provider>);
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check action on dispatching', () => {
    store.dispatch(formInputPseudo('bob'));
    const action = store.getActions();
    expect(action[0].type).toBe('FORM_INPUT_PSEUDO');
    expect(action[0].pseudo).toBe(initialState.userData.pseudo);
  });

  it('check onChange in pseudo input', () => {
    const input = wrapper.find('#pseudo');
    expect(input.props().value).toBe(initialState.userData.pseudo);

    const event = { target: { value: 'roger' } };
    input.props().onChange(event);
    const action = store.getActions();
    expect(action[0].type).toBe('FORM_INPUT_PSEUDO');
    expect(action[0].pseudo).toBe('roger');
  });
});
