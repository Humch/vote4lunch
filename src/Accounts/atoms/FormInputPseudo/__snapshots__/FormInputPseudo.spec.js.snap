// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputPseudo Snapshot capturing Snapshot of FormInputPseudo 1`] = `
<input
  className="form-control "
  id="pseudo"
  name="pseudo"
  onChange={[Function]}
  required={true}
  type="text"
  value="bob"
/>
`;
